### DISCUSSION TITLE
怎么挂载 mega 网盘？（怎么为 Alist 设置翻墙代理？）

### DISCUSSION THREAD

**Discussion by Skylarzz** [AlistGo/alist#4363] [General]
mega 网盘连接超时报错 `Post "https://g.api.mega.co.nz/cs?id=1234567": dial tcp 133.33.33.233:443: i/o timeout`
估计是需要翻墙，但是没找到为 Alist 设置翻墙代理的方法
求解求解，谢谢

**Comment by anwen-anyi** [comment]
https://anwen-anyi.github.io/index/07-wenti.html#_41-alist如何-使用-吃到-代理-proxy

**Reply by Skylarzz** [reply]
@anwen-anyi 
谢谢，可以使用了
使用类似方法为 cmd 设置了代理

具体过程如下，供大家参考
1. 新建文件 AlistStart.vbs（与 alist.exe 相同目录），这一步是为了隐藏 bat 的运行窗口
```
Dim ws
Set ws = Wscript.CreateObject("Wscript.Shell")
ws.run "AlistStart_StepII.bat",vbhide
Wscript.quit
```
2. 新建文件 AlistStart_StepII.bat，这一步是为了为 Alist 设置梯子
```
set http_proxy=http://127.0.0.1:7890
set https_proxy=http://127.0.0.1:7890
set no_proxy=localhost,127.0.0.1,example.com
start AlistStart_StepIII.vbs
```
3. 新建文件 AlistStart_StepIII.vbs，这一步是为了隐藏 Alist 的运行窗口
```
Dim ws
Set ws = Wscript.CreateObject("Wscript.Shell")
ws.run "alist.exe server",vbhide
Wscript.quit
```
4. 启用 Alist 时只要运行 AlistStart.vbs 即可

**Reply by anwen-anyi** [reply]
哈哈 开始CMD不行是因为 `export` 没办法在CMD输入语法不对 CMD 确实可以用你的 `set`来导入

**Reply by anwen-anyi** [reply]
我也看了 `set no_proxy=localhost,127.0.0.1,example.com` 只是 `localhost` `127.0.0.1` `example.com` 不走代理
那么和 `all_proxy=socks5://127.0.0.1:10809` 全部都走代理有什么区别吗？ 哈哈~ 

我当时用的是 `Git Bash`，这样是单独只让`AList`的文件夹走了代理好像？而且这个页面关闭了下次继续用的话还得重新输入才行
不知道`CMD`的这样设置会不会也让本机其他软件是走了代理？ ~~如果停止了`AList` 代理还会不会在~~ (验证了一下也是只能重新启动 哈哈 很好~

---- 
综上所述你这样也很好~~ 拿走补充去了 哈哈

**Reply by Javabcd** [reply]
我是这样处理的 windows版  原来我是用一个bat脚本启动alist的:
`alist server`
现在我把bat脚本改成这样就可以了:
```
set http_proxy=http://127.0.0.1:10809
set https_proxy=http://127.0.0.1:10809
set no_proxy=localhost,127.0.0.1
alist server
```
ps:我用的是v2rayN所以http端口是10809

**Comment by AliciaLEO** [comment]
### alist lunix（openwrt/istore）端走代理方法补充：
在/etc/**init.d目录**下找到alist文件，用记事本编辑该文件，更改如下：


找到以下代码：
```	
	procd_open_instance
	procd_set_param command "$PROG"
	procd_append_param command server
	procd_append_param command --data "$CONF_DIR"
	procd_append_param command --no-prefix

```


将其修改为：
```	
	procd_open_instance
	# --- 添加以下两行来设置代理环境变量 ---
	# 请将自己的代理主机地址替换为你的代理IP地址，请将自己设置的代理端口替换为你的代理端口
	# 如：http://192.168.100.1:10808
	procd_set_param env http_proxy="http://[自己的代理主机地址]:代理端口"
	procd_set_param env https_proxy="http://[自己的代理主机地址]:代理端口"
	# ------------------------------------
	procd_set_param command "$PROG"
	procd_append_param command server
	procd_append_param command --data "$CONF_DIR"
	procd_append_param command --no-prefix
	
```
**！！！！！然后重启alist，即可实现代理！！！！！**

**实测开启代理后Terabox已可用，可以正常上传，下载其云盘内容，不在报错：terabox is not yet available in this area**
